### PR TITLE
Simplify the status LED to save power

### LINKED ISSUE #4133: Revamp status LED signaling
@tannewt discussed revisiting the multi-colored status LED blinking that CIrcuitPython does to indicate errors. I think the ideas were to:
- simplify the indications
- do not repeat them indefinitely

@tannewt Please revise this post or comment as necessary.

### PR DESCRIPTION
**Pull request by tannewt**
This also removes the need to pin share because we don't use the
status LED while user code is running.

The status flashes fallback to the HW_STATUS LED if no RGB LED is
present. Each status has a unique blink pattern as well.

One caveat is the REPL state. In order to not pin share, we set the
RGB color once. PWM and single color will be shutoff immediately but
DotStars and NeoPixels will hold the color until the user overrides
it.

Fixes #4133

### CODE REVIEW

File: ports/nrf/boards/feather_nrf52840_express/mpconfigboard.h [hunk 1]
```diff
@@ -30,7 +30,7 @@
 #define MICROPY_HW_BOARD_NAME       "Adafruit Feather nRF52840 Express"
 #define MICROPY_HW_MCU_NAME         "nRF52840"
 
-#define MICROPY_HW_NEOPIXEL         (&pin_P0_16)
+// #define MICROPY_HW_NEOPIXEL         (&pin_P0_16)
```
Comment (Neradoc) [inline ports/nrf/boards/feather_nrf52840_express/mpconfigboard.h:33]: That seems unintentional.

Comment (tannewt) [reply ports/nrf/boards/feather_nrf52840_express/mpconfigboard.h:33]: Good catch! Thank you.

**Comment by tannewt** [comment]
> On a Feather M4 and Feather M0 Express, the LED stays either black or lit. It comes back to life when doing things like opening the drive in the Finder or activating and sending data to CDC2.

Thanks for finding this! I'm able to reproduce it but can't find the issue. Will have to get the debugger out tomorrow.

**Comment by jposada202020** [comment]
Led does not turn on
was tested on
```python
Adafruit CircuitPython 7.0.0-alpha.2-573-g49bad0797 on 2021-05-14; Adafruit Feather M4 CAN with same51j19a
```
Neopixel only turns green on bootoader mode.

**Comment by tannewt** [comment]
> Led does not turn on
> was tested on
> 
> ```python
> Adafruit CircuitPython 7.0.0-alpha.2-573-g49bad0797 on 2021-05-14; Adafruit Feather M4 CAN with same51j19a
> ```
> 
> Neopixel only turns green on bootoader mode.

I think this was broken before this change. The M4 CAN has a neopixel power pin that the core doesn't know how to turn on. On some boards with turn it off and on with `board_init` and `board_deinit` but I don't see that code for the M4 CAN.

**Comment by jposada202020** [comment]
Tested on
```python
Adafruit CircuitPython 7.0.0-alpha.2-608-gd134e2af8 on 2021-05-19; Adafruit Feather M4 CAN with same51j19a
```

### Results
#### REPL : NO light
#### Init : 3 Yellow - 1 Green
#### Code Running: 1 Green every 5-6 sec
####  Code Error: 2 Reds 
#### Bootloader: Green all the time

**Review by kattni — APPROVED** [review]
Tested final change on MagTag. LEDs are acting as expected - purple, three yellow blinks on startup, and blinking three yellows when in safe mode.

**Comment by Neradoc** [comment]
We can PR them after this one is merged, but there are 2 more boards that need their `MICROPY_HW_LED_[RGB]` changed to `CIRCUITPY_RGB_STATUS_[RGB]`:
- pimoroni_tiny2040 (also `CIRCUITPY_RGB_STATUS_INVERTED_PWM`)
- pimoroni_picosystem (probably inverted too ?)

**Review by kattni — APPROVED** [review]
Tested safe mode on ItsyBitsy, it now works properly. LEDs are intermittently flashing yellow as expected. Flashing on startup was also as expected.

**Pull request merged**